esmvaltool/cmorizers/data/formatters/datasets/merra.ncl

```diff
@@ -0,0 +1,306 @@
+  SOURCEFILE = (/"instM_3d_asm_Cp.", "instM_3d_asm_Cp.", \  ; 3d asm
+                 "instM_3d_asm_Cp.", "instM_3d_asm_Cp.", \
+                 "instM_3d_asm_Cp.", "instM_3d_asm_Cp.", \
+                 "instM_3d_asm_Cp.", "instM_3d_asm_Cp.", \
+                 "instM_3d_asm_Cp.", "instM_3d_asm_Cp.", \
+                 "instM_3d_asm_Cp.",                     \
+                 "instM_2d_int_Nx.", "instM_2d_int_Nx.", \  ; 2d int
+                 "instM_2d_int_Nx.",                     \
+                 "tavgM_2d_rad_Nx.", "tavgM_2d_rad_Nx.", \  ; 2d rad
+                 "tavgM_2d_rad_Nx.", "tavgM_2d_rad_Nx.", \
+                 "tavgM_2d_rad_Nx.", "tavgM_2d_rad_Nx.", \
+                 "tavgM_2d_rad_Nx.",                     \
+                 "tavgM_2d_chm_Fx.", "tavgM_2d_chm_Fx."/)   ; 2d chm
+
+  ; dataset doi numbers
+  DOI = (/"10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \  ; 3d asm
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW",                          \
+          "10.5067/QL0PGBK2CYJS", "10.5067/QL0PGBK2CYJS",  \  ; 2d int
+          "10.5067/QL0PGBK2CYJS",                          \
+          "10.5067/6UX3EDUNVUFK", "10.5067/6UX3EDUNVUFK",  \  ; 2d rad
+          "10.5067/6UX3EDUNVUFK", "10.5067/6UX3EDUNVUFK",  \
+          "10.5067/6UX3EDUNVUFK", "10.5067/6UX3EDUNVUFK",  \
+          "10.5067/6UX3EDUNVUFK",                          \
+          "10.5067/IYDN3LNZ63UE", "10.5067/IYDN3LNZ63UE"/)    ; 2d chm
+
+  ; MIP
+  MIP = "Amon"
+
+  ; Frequency
+  FREQ = "mon"
+
+  ; CMOR table
+  base = getenv("cmor_tables")
+
+  CMOR_TABLE = getenv("cmor_tables") + \
+    (/"/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \  ; 3d asm
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP,                                \
+      "/cmip5/Tables/CMIP5_" + MIP, "/custom/CMOR_lwp.dat",        \  ; 2d int
+      "/cmip5/Tables/CMIP5_" + MIP,                                \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \  ; 2d rad
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP,                                \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP/)    ; 2d chm
+
+  ; Type
+  TYPE = "reanaly"
+
+  ; Version
+  VERSION = "5.2.0"
+
+  ; Global attributes
+  SOURCE = "https://goldsmr3.gesdisc.eosdis.nasa.gov/data/MERRA_MONTHLY/"
+  REF = "doi: "
```

Comment:
Would it be needed to add the doi here?

Reply:
I think that is fine, because the actual doi of the variable written to the output file will added at the time the file is written (that is because every variable migh have a different doi). I can see that this is confusing and moved the definition of `REF` to where the output file is actually written.